packages/cactus-plugin-keychain-azure-kv/src/main/typescript/plugin-keychain-azure-kv.ts

```diff
@@ -0,0 +1,257 @@
+        opts.azureInMemoryCredentials.azurePassword,
+      );
+      this.azureKvClientMock = new PluginKeychainAzureKvMock({
+        azureKvUrl: this.vaultUrl,
+        azureCredentials: azureCredentials,
+      });
+    } else {
+      const azureCredentials = new DefaultAzureCredential();
+      this.azureKvClientMock = new PluginKeychainAzureKvMock({
+        azureKvUrl: this.vaultUrl,
+        azureCredentials: azureCredentials,
+      });
+    }
+
+    this.secrets = new Map<string, string>();
+
+    this.log.info(`Created ${this.className}. KeychainID=${opts.keychainId}`);
+  }
+
+  async registerWebServices(app: Express): Promise<IWebServiceEndpoint[]> {
+    const webServices = await this.getOrCreateWebServices();
+    await Promise.all(webServices.map((ws) => ws.registerExpress(app)));
+    return webServices;
+  }
+
+  public async getOrCreateWebServices(): Promise<IWebServiceEndpoint[]> {
+    if (Array.isArray(this.endpoints)) {
+      return this.endpoints;
+    }
+    const endpoints: IWebServiceEndpoint[] = [];
+
+    // TODO: Writing the getExpressRequestHandler() method for
+    // GetKeychainEntryEndpointV1 and SetKeychainEntryEndpointV1
+
+    // {
+    //   const ep = new GetKeychainEntryEndpointV1({
+    //     logLevel: this.opts.logLevel,
+    //   });
+    //   ep.registerExpress(expressApp);
+    //   endpoints.push(ep);
+    // }
+    // {
+    //   const ep = new SetKeychainEntryEndpointV1({
+    //     logLevel: this.opts.logLevel,
+    //   });
+    //   ep.registerExpress(expressApp);
+    //   endpoints.push(ep);
+    // }
+    this.endpoints = endpoints;
+
+    return endpoints;
+  }
+
+  public getHttpServer(): Optional<Server | SecureServer> {
+    return Optional.empty();
+  }
+
+  public async shutdown(): Promise<void> {
+    throw new Error("Method not implemented.");
+  }
+
+  public getInstanceId(): string {
+    return this.instanceId;
+  }
+
+  public getKeychainId(): string {
+    return this.opts.keychainId;
+  }
+
+  public getPackageName(): string {
+    return `@hyperledger/cactus-plugin-keychain-vault`;
+  }
+
+  public getAspect(): PluginAspect {
+    return PluginAspect.KEYCHAIN;
+  }
+
+  async rotateEncryptionKeys(): Promise<void> {
+    throw new Error("Method not implemented.");
+  }
+
+  public getEncryptionAlgorithm(): string {
+    return null as any;
+  }
+
+  public getAzureKvClient(): SecretClient {
+    return this.azureKvClientMock;
+  }
+
+  /**
+   * TO DO: Use SecretClient class of "@azure/keyvault-secrets" to implement this function
+   * instead of calling the mock implementation, which stores the secrets In-Memory
+   */
+  async get<T>(key: string): Promise<T> {
+    const keyVaultSecret: KeyVaultSecret = await this.azureKvClientMock.getSecret(
+      key,
+    );
+    if (keyVaultSecret) {
+      const result = keyVaultSecret.value;
+      return (result as unknown) as T;
+    } else {
+      return (false as unknown) as T;
```

Comment:
@jagpreetsinghsasan Throw if it doesn't exist. Then the endpoints can return HTTP 404 upstream and we can write tests asserting that they do (later)

Reply:
Actually, I initially had a throw statement when the key doesnt exist.
But then in the test case, it fails like this
![image](https://user-images.githubusercontent.com/56861721/119941563-671b7700-bfae-11eb-909c-f30bea56f9a2.png)

Reply:
@jagpreetsinghsasan That's a good thing IMO, you can make the test work then by either asserting that the promise had been rejected (via `promise-tape`'s `rejects(...)` method) or just doing an old school try-catch and then asserting that the catch block was executed.

Reply:
Oh, where did it go? XD
I added the throws here and added the try catch block in test, now its gone from here, but it is there.
I am fixing this, along with the other review comment about the assertion